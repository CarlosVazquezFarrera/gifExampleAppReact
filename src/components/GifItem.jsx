import PropTypes from "prop-types";
export default function GifItem({ title, url }) {
  return (
    <div className="card">
      <img src={url} alt={title} />
    </div>
  );
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
