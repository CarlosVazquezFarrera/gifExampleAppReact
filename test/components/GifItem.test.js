import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("GifItem", () => {
  const url =
    "https://media4.giphy.com/media/fwii0mKnvHshW/giphy.gif?cid=26e8e2e09fnihkdk5c3gp8ib5izhjbgs9ajrpfbuogtb8q7e&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  const title = "test";
  test("should match snatshop", () => {
    const { container } = render(<GifItem url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("should math alt and src", () => {
    render(<GifItem url={url} title={title} />);
    const { src, alt } = screen.getByRole("img");

    expect(src).toEqual(url);
    expect(alt).toEqual(title);
  });
});
