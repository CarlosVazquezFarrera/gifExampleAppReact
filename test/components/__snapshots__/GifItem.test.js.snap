// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match snatshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://media4.giphy.com/media/fwii0mKnvHshW/giphy.gif?cid=26e8e2e09fnihkdk5c3gp8ib5izhjbgs9ajrpfbuogtb8q7e&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
